### DISCUSSION TITLE
The use of the functions of  opTomo

### DISCUSSION THREAD

**Discussion by scf819** [astra-toolbox/astra-toolbox#528] [Q&A] [answered]
Create the Spot operator for ASTRA using the GPU.
`A = opTomo('cuda', proj_geom, vol_geom);`
The data type of variable A is optomo;

Can I get data  of A from Spot operator?

I want to use the data of A. It seems that Some  functions don't accept data of the optomotype.

**Comment by askorikov** [comment]
Hi! I'm not sure I completely get the question. What do you mean by "data of A"?

**Reply by scf819** [reply]
s017_opTomo.m  ：

% load a phantom image
im = phantom(128);
% and flatten it to a vector
x = im(:);

%% Setting up the geometry
% projection geometry
proj_geom = astra_create_proj_geom('parallel', 1, 128, linspace2(0,pi,180));
% object dimensions
vol_geom  = astra_create_vol_geom(128,128);

%% Generate projection data
% Create the Spot operator for ASTRA using the GPU.
W = opTomo('cuda', proj_geom, vol_geom);

p = W*x;

The W size should be 23040*46384；The data type is opTpmo；
I want to get the single or double data of W，size is 23040*46384；
The current way I get the data of W：
ss=ones(1,23040);
z=W(:,1);
pp=z*ss
sucf=diag(pp);
This gives us a column of data for W of single。
I don't know if there's a function that I can get W in one step.

**Reply by askorikov** [reply, marked as answer]
I see. If you want to get the full projection matrix, you can look into this example: https://github.com/astra-toolbox/astra-toolbox/blob/master/samples/matlab/s009_projection_matrix.m
